clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2834,6 +2885,14 @@ private int internalProcessConsumerRecord(
           LOGGER.error("Failed to record Record heartbeat with message: ", e);
         }
       } else {
+        // TODO: This is a hack. Today the code kind of does a backdoor change to a key in the leaderProducer callback.
```

Comment:
Actually I don't really follow this hack, can you explain a bit? I remember the issue being the leader does not chunk the record before it is produced to VT, and it seems like here it does not solve the issue? So the Leader and the Follower will still have different view?